os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -279,27 +277,21 @@ static inline int imxrt_bulkerase(struct imxrt_dev_s *priv, unsigned long arg)
 	/* Deselect the FLASH */
 
 	SPI_SELECT(priv->dev, SPIDEV_FLASH, false);
+#endif
 	
 	size_t sectors_erased = 0;
-	size_t startsector = 8;
-//	fdbg("startsector: %d nsectors: %d\n", (long)startsector, (int)priv->nsectors);
-	fdbg("startsector: %d nsectors: %d\n", (long)startsector, 2);
+	fvdbg("priv: %p, startblock: %d, startaddr: %x, nsectors: %d\n", priv, startblock, startblock >> priv->pageshift, nsectors);
 
 	/* Lock access to the SPI bus until we complete the erase */
 
-//	while (sectors_erased < priv->nsectors) {
-	while (sectors_erased < 2) {
+	while (sectors_erased < nsectors) {
 		/* Not using sub-sector erase.  Erase each full sector */
 
-		imxrt_sectorerase(priv, startsector + sectors_erased);
+		imxrt_sectorerase(priv, (startblock >> priv->pageshift) + sectors_erased);
 		sectors_erased++;
 	}
+	fvdbg("ret = %d, erased sectors = %d\n", ret, sectors_erased);
```

Comment:
Is `ret` declared or used anywhere in this function? Please check and ignore comment if unnecessary.

Reply:
That's a good points, modified.